### PR TITLE
Refactor/matrix room client slice

### PR DESCRIPTION
**Pull request by radaljj**
### Summary

Refactors Matrix room-related operations from `MatrixSynapseService` into a dedicated `MatrixRoomClient`.

### Changes

- Extracted Matrix room operations into `MatrixRoomClient`.
- Kept existing `MatrixSynapseService` public methods unchanged.
- Preserved existing behavior and request flow.
- Added regression tests for success, failure and edge-case room scenarios.

### Covered Operations

- Create Matrix room
- Invite user to room
- Join room
- Remove user from room
- Update room power levels

### Testing

- `MatrixRoomClientTest`
- `compile`
- `spotless:check`
- `git diff --check`

### Notes

Internal refactor only. No API, database or functional behavior change intended.

### CODE REVIEW

File: src/main/java/de/caritas/cob/userservice/api/adapters/matrix/MatrixRoomClient.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  }
+
+  public boolean joinRoom(String roomId, String accessToken) {
+    try {
+      var headers = getClientHttpHeaders(accessToken);
+      headers.setContentType(MediaType.APPLICATION_JSON);
+
+      HttpEntity<String> request = new HttpEntity<>("{}", headers);
+
+      var url = matrixConfig.getApiUrl(ENDPOINT_JOIN_ROOM.replace("{roomId}", roomId));
```
Comment (shazia-k) [inline src/main/java/de/caritas/cob/userservice/api/adapters/matrix/MatrixRoomClient.java]: Same as above.

File: src/test/java/de/caritas/cob/userservice/api/adapters/matrix/MatrixRoomClientTest.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+  @Test
+  void removeUserFromRoom_ShouldReturnFalse_WhenMatrixRejectsMembershipUpdate() {
+    doThrow(new RuntimeException("remove failed"))
+        .when(restTemplate)
+        .put(
+            eq(API_URL + "/_matrix/client/r0/rooms/" + ROOM_ID + "/state/m.room.member/" + USER_ID),
+            org.mockito.ArgumentMatchers.any(HttpEntity.class));
+
+    assertThat(matrixRoomClient.removeUserFromRoom(ROOM_ID, USER_ID, ACCESS_TOKEN)).isFalse();
+  }
```
Comment (shazia-k) [inline src/test/java/de/caritas/cob/userservice/api/adapters/matrix/MatrixRoomClientTest.java:374]: Issues:

- The generic catch (Exception) branch in createRoom and inviteUserToRoom is never tested.

- HttpClientErrorException on the GET step of setUserPowerLevel is untested (only the PUT step is covered).

- HttpClientErrorException path in removeUserFromRoom is untested.

- A 5xx / non-2xx response from joinRoom returning false is not tested.

Please add missing tests

File: src/main/java/de/caritas/cob/userservice/api/adapters/matrix/MatrixRoomClient.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        log.error("Failed to get current power levels for room {}", roomId);
+        return false;
+      }
+
+      @SuppressWarnings("unchecked")
+      Map<String, Object> powerLevels = new HashMap<>(currentResponse.getBody());
+
+      @SuppressWarnings("unchecked")
+      Map<String, Integer> users =
+          (Map<String, Integer>) powerLevels.getOrDefault("users", new HashMap<>());
```
Comment (shazia-k) [inline src/main/java/de/caritas/cob/userservice/api/adapters/matrix/MatrixRoomClient.java]: Unsafe cast (Map<String, Integer>) — the response body is a raw Map, so Jackson may return Long/Double values for large numbers; the cast succeeds at compile time but can throw a ClassCastException at runtime, silently swallowed by the generic catch.

File: src/main/java/de/caritas/cob/userservice/api/adapters/matrix/MatrixRoomClient.java [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+          "Failed to set power level for user {} in room {}: {}", userId, roomId, e.getMessage());
+      return false;
+    }
+  }
+
+  public boolean removeUserFromRoom(String roomId, String userId, String accessToken) {
+    try {
+      String url =
+          matrixConfig.getApiUrl(
+              ENDPOINT_MEMBERSHIP.replace("{roomId}", roomId).replace("{userId}", userId));
```
Comment (shazia-k) [inline src/main/java/de/caritas/cob/userservice/api/adapters/matrix/MatrixRoomClient.java]: Follow the 1st comment to get API url

**Review by shazia-k — CHANGES_REQUESTED** [review]
@radaljj 
I leave some comments which needs to be fixed. Can you please address those?

**Comment by radaljj** [comment]
> @radaljj I leave some comments which needs to be fixed. Can you please address those?

Sure no problem i will fix it now

**Review by shazia-k — APPROVED** [review]
